PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = "hr.attendance"
-    _inherit = 'hr.leave'
-    _inherit = "hr.leave.accrual.level"
-    _inherit = 'hr.leave.allocation'
-    _inherit = 'hr.leave.type'
-    _inherit = 'res.users'

Total Changes: 22

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_holidays_attendance/models/hr_attendance.py
+++ b/addons/hr_holidays_attendance/models/hr_attendance.py
@@ -6,7 +6,7 @@ from odoo.osv.expression import AND
 
 
 class HrAttendance(models.Model):
-    _inherit = "hr.attendance"
+    _inherit = ["hr.attendance"]
 
     def _get_overtime_leave_domain(self):
         domain = super()._get_overtime_leave_domain()

--- a/addons/hr_holidays_attendance/models/hr_leave.py
+++ b/addons/hr_holidays_attendance/models/hr_leave.py
@@ -9,8 +9,8 @@ from odoo.exceptions import ValidationError
 from odoo.tools import float_round
 
 
-class HRLeave(models.Model):
-    _inherit = 'hr.leave'
+class HrLeave(models.Model):
+    _inherit = ['hr.leave']
 
     overtime_id = fields.Many2one('hr.attendance.overtime', string='Extra Hours')
     employee_overtime = fields.Float(related='employee_id.total_overtime', groups='base.group_user')

--- a/addons/hr_holidays_attendance/models/hr_leave_accrual_plan_level.py
+++ b/addons/hr_holidays_attendance/models/hr_leave_accrual_plan_level.py
@@ -4,8 +4,8 @@
 from odoo import api, fields, models
 
 
-class AccrualPlanLevel(models.Model):
-    _inherit = "hr.leave.accrual.level"
+class HrLeaveAccrualLevel(models.Model):
+    _inherit = ["hr.leave.accrual.level"]
 
     frequency_hourly_source = fields.Selection(
         selection=[

--- a/addons/hr_holidays_attendance/models/hr_leave_allocation.py
+++ b/addons/hr_holidays_attendance/models/hr_leave_allocation.py
@@ -8,8 +8,8 @@ from odoo.exceptions import ValidationError
 from odoo.osv import expression
 
 
-class HolidaysAllocation(models.Model):
-    _inherit = 'hr.leave.allocation'
+class HrLeaveAllocation(models.Model):
+    _inherit = ['hr.leave.allocation']
 
     def default_get(self, fields):
         res = super().default_get(fields)

--- a/addons/hr_holidays_attendance/models/hr_leave_type.py
+++ b/addons/hr_holidays_attendance/models/hr_leave_type.py
@@ -5,8 +5,8 @@ from odoo.tools.misc import format_duration
 from odoo import _, api, fields, models
 
 
-class HRLeaveType(models.Model):
-    _inherit = 'hr.leave.type'
+class HrLeaveType(models.Model):
+    _inherit = ['hr.leave.type']
 
     overtime_deductible = fields.Boolean(
         "Deduct Extra Hours", default=False,
@@ -30,7 +30,7 @@ class HRLeaveType(models.Model):
                 'count': _('%s hours available',
                     format_duration(employee.total_overtime)),
             }
-        super(HRLeaveType, self - overtime_leaves)._compute_display_name()
+        super(HrLeaveType, self - overtime_leaves)._compute_display_name()
 
     def get_allocation_data(self, employees, date=None):
         res = super().get_allocation_data(employees, date)

--- a/addons/hr_holidays_attendance/models/res_users.py
+++ b/addons/hr_holidays_attendance/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResUsers(models.Model):
-    _inherit = 'res.users'
+    _inherit = ['res.users']
 
     request_overtime = fields.Boolean(compute='_compute_request_overtime')
 
